ksql-engine/src/main/java/io/confluent/ksql/function/GeneratedAggregateFunction.java

```diff
@@ -166,7 +169,22 @@ public A aggregate(final V currentValue, final A aggregateValue) {
     };
   }
 
-  protected static <V,A> Supplier<A> supplier(final Udaf<V, A> udaf) {
+  @Override
+  public ValueMapper<A, O> getMapper() {
+    return (v1) -> {
+      final long start = Time.SYSTEM.nanoseconds();
+      try {
+        //noinspection unchecked
+        return udaf.map(v1);
+      } finally {
+        if (aggregateSensor != null) { //TODO add sensor for mapper?
```

Comment:
i agree with your TODO - I don't think it makes sense to use the aggregate sensor here